pkg/gateway/gateway.go

```diff
@@ -0,0 +1,122 @@
+package gateway
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"io"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/recover"
+)
+
+var syntheticError = errors.New("synthetic error")
```

Comment:
Please explain the name? :)

Reply:
Will add comments for this on next push.